### PR TITLE
chore: Only publish relevant files to npm

### PR DESCRIPTION
**Pull request by mattsjones**
Before:
image

After:
image

### CODE REVIEW

**Comment by maraisr** [comment]
Out of curiosity; is there a reason why that couldn't have been articulated with the `files` property in the `package.json`? Ie, only send files you want, rather than specifically exclude files/folders? Just thinking for the chance that a new root folder gets added, that forgets to be added to the npmignore.

cc @mattsjones

**Comment by mattsjones** [comment]
The `files` field would work as well. Just prefer a blacklist in this case though. I'd rather over-push to npm rather than forget to add a critical file/folder to the `files` whitelist.

File: .npmignore [hunk 1]
```diff
@@ -1 +1,15 @@
-demo/
+.github/
+docs/
+generate-component-docs/
+report/
+scripts/
+site/
+dist-storybook/
+icons/
```
Comment (markdalgleish) [inline .npmignore]: Are you sure this won't exclude our icon components?

Comment (mattsjones) [reply .npmignore]: Yep, checked that.

**Pull request merged**